class.jetpack-cli.php

```diff
@@ -1476,6 +1476,156 @@ private function partner_provision_error( $error ) {
+	 * --description: Allows to provide a text description of the block.
+	 * --keywords: Provide up to three keywords separated by comma so users  when they search for a block in the editor.
+	 *
+	 * ## BLOCK TYPE EXAMPLES
+	 *
+	 * wp jetpack scaffold block "Cool Block"
+	 * wp jetpack scaffold block "Amazing Rock" --slug="good-music" --description="Rock the best music on your site"
+	 * wp jetpack scaffold block "Jukebox" --keywords="music, audio, media"
+	 *
+	 * @subcommand scaffold block
+	 * @synopsis <type> <title> [--slug] [--description] [--keywords]
+	 *
+	 * @param array $args       Positional parameters, when strings are passed, wrap them in quotes.
+	 * @param array $assoc_args Associative parameters like --slug="nice-block".
+	 */
+	public function scaffold( $args, $assoc_args ) {
+		// It's ok not to check if it's set, because otherwise WPCLI exits earlier.
+		switch ( $args[0] ) {
+			case 'block':
+				$this->block( $args, $assoc_args );
+				break;
+			default:
+				WP_CLI::error( sprintf( esc_html__( 'Invalid subcommand %s.', 'jetpack' ), $args[0] ) . ' 👻' );
+				exit( 1 );
+		}
+	}
+
+	/**
+	 * Creates the essential files in Jetpack to build a Gutenberg block.
+	 *
+	 * @param array $args       Positional parameters. Only one is used, that corresponds to the block title.
+	 * @param array $assoc_args Associative parameters defined in the scaffold() method.
+	 */
+	public function block( $args, $assoc_args ) {
+		if ( isset( $args[1] ) ) {
+			$title = $args[1];
+		} else {
+			WP_CLI::error( esc_html__( 'The title parameter is required.', 'jetpack' ) . ' 👻' );
+			exit( 1 );
+		}
+
+		$slug = isset( $assoc_args['slug'] )
+			? $assoc_args['slug']
+			: sanitize_title( $title );
+
+		if ( ! preg_match( '/^[a-z][a-z0-9\-]*$/', $slug ) ) {
+			WP_CLI::error( esc_html__( 'Invalid block slug. They can contain only lowercase alphanumeric characters or dashes, and start with a letter', 'jetpack' ) . ' 👻' );
+		}
+
+		global $wp_filesystem;
+		if ( ! WP_Filesystem() ) {
+			WP_CLI::error( esc_html__( "Can't write files", 'jetpack' ) . ' 😱' );
+		}
+
+		$path = JETPACK__PLUGIN_DIR . "extensions/blocks/$slug";
+
+		if ( $wp_filesystem->exists( $path ) && $wp_filesystem->is_dir( $path ) ) {
+			WP_CLI::error( sprintf( esc_html__( 'Name conflicts with the existing block %s', 'jetpack' ), $path ) . ' ⛔️' );
+			exit( 1 );
+		}
+
+		$wp_filesystem->mkdir( $path );
+
+		$hasKeywords = isset( $assoc_args['keywords'] );
+
+		$files = array(
+			"$path/$slug.php" => $this->render_block_file( 'block-register-php', array(
+				'slug' => $slug,
+				'title' => $title,
+				'underscoredSlug' => str_replace( '-', '_', $slug ),
+			) ),
+			"$path/index.js" => $this->render_block_file( 'block-index-js', array(
+				'slug' => $slug,
+				'title' => $title,
+				'description' => isset( $assoc_args['description'] )
+					? $assoc_args['description']
+					: $title,
+				'keywords' => $hasKeywords
```

Comment:
Should we add a check for the number of keywords provided here, and error if someone provides too many?

Reply:
Yeah, makes sense

Reply:
I'll limit the number of keywords accepted to 3 and dismiss the rest

```php
explode( ',', $assoc_args['keywords'], 3 )
```